docs/12-self-hosting/03-host-creation/03-QEMU/02-linux-cloudimage.mdx

```diff
@@ -0,0 +1,199 @@
+
+  # Number of physical CPU cores to allocate to the VM
+  export PHYSICAL_CORES="2"
+
+  # Number of threads per core.
+  # Set this to `1` for CPUs that do not support hyper-threading
+  export THREADS="1"
+  export SMP=$(( $PHYSICAL_CORES * $THREADS ))
+
+  # Amount of Disk Space to allocate to the VM.
+  # Cannot exceed available on host.
+  export DISK_SIZE="32G"
+
+  # Amount of RAM to allocate to the VM.
+  # Cannot exceed available RAM on host.
+  export MEMORY="8G"
+
+  # IP address where host may be reached. Do not use `localhost`.
+  export HOST_ADDRESS="SOME IP HERE"
+
+  # Port used by SSH on the host
+  export HOST_SSH_PORT="22"
+
+  # Port to use when forwarding SSH to the VM
+  export VM_SSH_PORT="1234"
+
+  # Port number to expose on the host for VNC
+  export VNC_PORT="0"
+  ```
+
+### Credentials
+
+- Create an SSH Key
+
+  ```bash
+  yes |ssh-keygen -C "$VM_USER" \
+      -f runner \
+      -N '' \
+      -t rsa
+  ```
+
+- Create a password
+
+  ```bash
+  # Install the mkpasswd utility
+  sudo apt install -y whois
+
+  read PW_STRING
+  export PASSWORD=$(mkpasswd -m sha-512 --rounds=4096 "$PW_STRING" -s "saltsaltlettuce")
+  ```
+
+### Cloud-init Config
+
+- Create a Cloud-Init file
+
+  See the [Cloud-Init](https://cloudinit.readthedocs.io/en/latest/topics/examples.html) official
+  docs for more information about Cloud-Init. More advanced templates are available in the
+  [Host Provisioning](../../04-host-provisioning/03-Cloud-Init.mdx) directory.
+
+  ```bash
+  VM_KEY=$(cat runner.pub)
+
+  /bin/cat << EOF > cloud-init.yaml
+  #cloud-config
+  hostname: runner
+  disable_root: false
+  network:
+    config: disabled
+  users:
+    - name: ${VM_USER}
+      groups: users, admin, sudo
+      sudo: ALL=(ALL) NOPASSWD:ALL
+      shell: /bin/bash
+      lock_passwd: false
+      passwd: ${PASSWORD}
+      ssh_authorized_keys:
+        - ${VM_KEY}
+  EOF
+  ```
+
+### Disks
+
+- Create a Cloud-Init disk
+
+  ```bash
+  cloud-localds seed.img cloud-init.yaml
+  ```
+
+- Create a virtual disk using the cloud-image as a read-only backing file.
+
+  ```bash
+  qemu-img create -b ${CLOUD_IMAGE_NAME} -f qcow2 \
+      -F qcow2 disk.qcow2 \
+      "$DISK_SIZE" 1> /dev/null
+  ```
+
+## Create the VM
+
+- New guest
+
+  ```bash
+  sudo qemu-system-x86_64 \
+    -machine accel=kvm,type=q35 \
+    -cpu host \
+    -smp $SMP,sockets=1,cores="$PHYSICAL_CORES",threads="$THREADS",maxcpus=$SMP \
+    -m "$MEMORY" \
+    -serial stdio -vga virtio -parallel none \
+    -device virtio-net-pci,netdev=network \
+    -netdev user,id=network,hostfwd=tcp::"${VM_SSH_PORT}"-:"${HOST_SSH_PORT}" \
+    -object iothread,id=io \
+    -device virtio-blk-pci,drive=disk,iothread=io \
+    -drive if=none,id=disk,cache=none,format=qcow2,aio=threads,file=disk.qcow2 \
+    -drive if=virtio,format=raw,file=seed.img,index=0,media=disk \
+    -bios /usr/share/ovmf/OVMF.fd \
+    -usbdevice tablet \
+    -vnc $HOST_ADDRESS:$VNC_PORT
+  ```
```

Comment:
Not 100% sure about this here, untested, but is it better to have double quotes for all of the env vars?

```bash
sudo qemu-system-x86_64 \
    -machine accel=kvm,type=q35 \
    -cpu host \
    -smp "$SMP",sockets=1,cores="$PHYSICAL_CORES",threads="$THREADS",maxcpus="$SMP" \
    -m "$MEMORY" \
    -serial stdio -vga virtio -parallel none \
    -device virtio-net-pci,netdev=network \
    -netdev user,id=network,hostfwd=tcp::"$VM_SSH_PORT"-:"$HOST_SSH_PORT" \
    -object iothread,id=io \
    -device virtio-blk-pci,drive=disk,iothread=io \
    -drive if=none,id=disk,cache=none,format=qcow2,aio=threads,file=disk.qcow2 \
    -drive if=virtio,format=raw,file=seed.img,index=0,media=disk \
    -bios /usr/share/ovmf/OVMF.fd \
    -usbdevice tablet \
    -vnc "$HOST_ADDRESS":"$VNC_PORT"
```

Reply:
it *IS* a best practice to double-quote, I'll be less lazy and do a check to see where I've missed them and get it added. 🙏

Reply:
added in commit https://github.com/game-ci/documentation/pull/419/commits/b7d5010b60f268c808ec22bc24798e92f1add086